src/lib/operators/join_hash.cpp

```diff
@@ -194,31 +194,32 @@ std::shared_ptr<const Table> JoinHash::_on_execute() {
           !std::is_same_v<pmr_string, BuildColumnDataType> && !std::is_same_v<pmr_string, ProbeColumnDataType>;
 
       if constexpr (BOTH_ARE_STRING || NEITHER_IS_STRING) {
-        if (!_radix_bits) {
+        if (!_radix_bits.load()) {
```

Comment:
So the concurrency issues come from the fact that this variable can be written while being copied? Why do we even copy this? The number of radix bits depends on the build side's input side and this size also changes when the query is cached if there are changes to the data.